src/settings/settingsManager.ts

```diff
@@ -1,40 +1,18 @@
-import { SettingsManager } from 'tauri-settings';
+import type { SettingsManager } from 'tauri-settings';
 import { Path, PathValue } from 'tauri-settings/dist/types/dot-notation';
 import { getDotNotation, setDotNotation } from 'tauri-settings/dist/utils/dot-notation';
 
-import { getSystemConfigurationsDir } from '../io/filesystem';
-import { readEnv } from '../io/readEnv';
+import { universalGet, universalPost } from '../api/api';
+import type * as apiTypes from '../api/api-types';
 
-export type OpenAiManner = 'concise' | 'elaborate' | 'scholarly';
+export type OpenAiManner = apiTypes.RewriteMannerType;
 export function getMannerOptions(): OpenAiManner[] {
   return ['concise', 'elaborate', 'scholarly'];
 }
 
-export interface SettingsSchema {
-  /**
-   * @deprecated this exists for retro-compatibility with v23.1.0 and should be removed.
-   */
-  general?: {
-    appDataDir: string;
-    projectName: string;
-  };
-  project: {
-    currentDir: string;
-  };
-  openAI: {
-    apiKey: string;
-    chatModel: string;
-    manner: OpenAiManner;
-    /** Control the "creativity" of OpenAI. Value should be between 0.7 and 0.9 */
-    temperature: number;
-  };
-  sidecar: {
-    logging: {
-      active: boolean;
-      path: string;
-    };
-  };
-}
+type DeepRequired<T> = T extends object ? { [k in keyof T]-?: DeepRequired<T[k]> } : T;
+
+export type SettingsSchema = DeepRequired<apiTypes.SettingsSchema>;
```

Comment:
Chatted about this briefly with @gjreda. This is going to be an issue with any Pydantic classes with default values, i.e. most of our current API types. These get modeled on the TS side as optional properties. For the settings PUT request body, I think this may be what we want (you don't have to set all the settings at once) but for the response type it isn't. All fields are required on the `/api/settings/` response. I've papered over that here but ultimately we'll want to have distinct request/response types coming from the Python side.